Use Symfony Messenger to handle ItemImport messages (#134)
webgriffe/SyliusAkeneoPlugin: README.md

### Launch data import from CLI

> To actually import data you must first create queue items with the **import command** and then you can import them with the **consume command**.

Reviewer: ```suggestion
To actually import data from CLI you can use the **import command** (see below).
```

```bash
bin/console webgriffe:akeneo:import --since="2020-01-30"
```

This will import all Akeneo entities updated after the provided date.

You can also use a "since file" where to read the since date:

```bash
echo "2020-01-30" > var/storage/akeneo-sincefile.txt
bin/console webgriffe:akeneo:import --since-file="var/storage/akeneo-sincefile.txt"
```

When run with the since file, the import command will write the current date/time to the since file after the importing process is terminated. This is useful when you put the import command in cron:

```
*  * * * * /usr/bin/php /path/to/sylius/bin/console -e prod -q webgriffe:akeneo:import --since-file=/path/to/sylius/var/storage/akeneo-import-sincefile.txt
```

This way the import command is run repeatedly importing only products modified since the last command execution.

You can also import items only for specific importers:

```bash
bin/console webgriffe:akeneno:import --importer="Product" --importer="MyImporter" --since="2020-01-30"
```

You can also import items regardless of their last update date:

```bash
bin/console webgriffe:akeneno:import --all
```

> #### Consume command

Reviewer: This section should be replaced with a statement about Symfony Messenger that must be in place and working to have data actually imported.

Author: With Sylius v1.11 Symfony Messenger should be already installed and configured. Anyway, a good portion of the doc should be rewritten.

Reviewer: Ok maybe it's enough to remove this "Consume command" paragraph and add an additional sentence to the previous one like:

> Import command assume that Symfony Messenger is working as required since Sylius v1.11.

Reviewer: We should also remove the "Browsing queue items in the admin" and "Queue clear" sections.

Author: Done!

To make all importers and other plugin features work automatically the following is the suggested crontab:

```
0   *   *  *  *  /path/to/sylius/bin/console -e prod -q webgriffe:akeneo:import --all --importer="AttributeOptions"
*   *   *  *  *  /path/to/sylius/bin/console -e prod -q webgriffe:akeneo:import --since-file=/path/to/sylius/var/storage/akeneo-import-sincefile.txt --importer="Product" --importer="ProductAssociations"

> *   *   *  *  *  /path/to/sylius/bin/console -e prod -q webgriffe:akeneo:consume
> 0   0   *  *  *  /path/to/sylius/bin/console -e prod -q webgriffe:akeneo:cleanup-queue

Reviewer: ```suggestion
```

0   */6 *  *  *  /path/to/sylius/bin/console -e prod -q webgriffe:akeneo:reconcile
```

This will:
* Import the update of all attribute options every hour
* Import, every minute, all products that have been modified since the last execution, along with their associations
* Clean the imported items queue older than 10 days, every day at midnight
* Reconcile Akeneo deleted products every 6 hours

> Import, Consume and Reconcile commands uses a [lock mechanism](https://symfony.com/doc/current/console/lockable_trait.html) which prevents running them if another instance of the same command is already running.

Reviewer: ```suggestion
Import and Reconcile commands uses a [lock mechanism](https://symfony.com/doc/current/console/lockable_trait.html) which prevents running them if another instance of the same command is already running.
```
